src/object/object.util.ts

```diff
@@ -507,3 +509,63 @@ export function _objectAssignExact<T extends AnyObject>(target: T, source: T): v
     }
   }
 }
+
+export function _diff<T extends AnyObject>(
+  before: T,
+  after: T,
+  options?: Partial<ObjectDiffOptions<T>>,
+): ObjectDiffResult<T> {
+  const { ignore } = { ..._diffDefaultOptions, ...options }
+  const keys = [..._objectKeys(before), ..._objectKeys(after)]
+  const diff: ObjectDiffResult<T> = {}
+
+  for (const k of keys) {
+    if (ignore.includes(k)) continue
+    if (_deepEquals(before[k], after[k])) continue
+
+    diff[k] = { from: before[k], to: after[k] }
```

Comment:
I guess in an ideal would it would be nicer to extract out the individual nested properties as opposed to the whole object, but can also be a future improvement if there was ever any desire. (It's a bit like how annoying it is with jest sometimes when it gives you 40 properties instead of only telling you the once that has change, I think when you use `toHaveBeenCalledWith()`)

Reply:
I am not sure I understand. 😬 

Can you provide an example result you would like to see coming from `_diff`?